src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -232,8 +232,20 @@ def confify(entries):
     # Ensure that the jar/rsc jar is on the rsc_mixed_compile_classpath.
     for target in targets:
       merged_cc = compile_contexts[target]
-      rsc_cc = merged_cc.rsc_cc
       zinc_cc = merged_cc.zinc_cc
+      rsc_cc = merged_cc.rsc_cc
+      # Make sure m.jar is digested if it exists when the target is validated.
+      if rsc_cc.rsc_jar_file.directory_digest is None and os.path.exists(rsc_cc.rsc_jar_file.path):
+        relpath = fast_relpath(rsc_cc.rsc_jar_file.path, get_buildroot())
+        classes_dir_snapshot, = self.context._scheduler.capture_snapshots([
```

Comment:
This is still a bandaid, because it might hide cases "other than" a cache hit. But as long as we're able to commit to starting JVM compile on v2 soon, it should be ok.

Reply:
Yeah, agreed, still not great. Although unless we are moving the logic to cache manager to snapshot everything relating to a set of tasks, this is likely the solution to go, i.e. after cache hit/target validation, we go into the target results dir and find the m.jar and digest it.